dataactbroker/handlers/fileHandler.py

```diff
@@ -636,17 +657,31 @@ def upload_fabs_file(self, fabs):
                 s3.upload_fileobj(fabs, bucket_name, filename_key)
             else:
                 fabs.save(filename_key)
-            return self.finalize(job_dict["fabs_id"])
+            json_response = self.finalize(job_dict["fabs_id"])
         except (ValueError, TypeError, NotImplementedError) as e:
-            return JsonResponse.error(e, StatusCode.CLIENT_ERROR)
+            json_response = JsonResponse.error(e, StatusCode.CLIENT_ERROR)
         except ResponseException as e:
             # call error route directly, status code depends on exception
-            return JsonResponse.error(e, e.status)
+            json_response = JsonResponse.error(e, e.status)
         except Exception as e:
             # unexpected exception, this is a 500 server error
-            return JsonResponse.error(e, StatusCode.INTERNAL_ERROR)
-        except:
-            return JsonResponse.error(Exception("Failed to catch exception"), StatusCode.INTERNAL_ERROR)
+            json_response = JsonResponse.error(e, StatusCode.INTERNAL_ERROR)
+        finally:
+            # handle a missing JSON response
+            if json_response is None:
+                json_response = JsonResponse.error(Exception("Failed to catch exception"), StatusCode.INTERNAL_ERROR)
+
+            if json_response.status_code != StatusCode.OK and submission:
+                jobs = sess.query(Job).filter(Job.submission_id == submission.submission_id,
+                                              Job.job_type_id == JOB_TYPE_DICT['file_upload'],
+                                              Job.job_status_id == JOB_STATUS_DICT['running'],
+                                              Job.file_type_id == FILE_TYPE_DICT_LETTER_ID['FABS']).all()
```

Comment:
Are we ever going to have more than one upload job for a given FABS submission? Do we need the `.all()` and the loop? If we feel that it isn't too bad to do this we can leave it, just seems odd.